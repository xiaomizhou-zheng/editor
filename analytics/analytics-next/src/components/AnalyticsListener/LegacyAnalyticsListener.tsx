import React, { Component } from 'react';

import PropTypes from 'prop-types';

import {
  default as AnalyticsReactContext,
  AnalyticsReactContextInterface,
} from '@atlaskit/analytics-next-stable-react-context';

import UIAnalyticsEvent, {
  UIAnalyticsEventHandler,
} from '../../events/UIAnalyticsEvent';

type Props = {
  /** Children! */
  children?: React.ReactNode;
  /** The channel to listen for events on. */
  channel?: string;
  /** A function which will be called when an event is fired on this Listener's
   * channel. It is passed the event and the channel as arguments. */
  onEvent: (event: UIAnalyticsEvent, channel?: string) => void;
};

const ContextTypes = {
  getAtlaskitAnalyticsEventHandlers: PropTypes.func,
  getAtlaskitAnalyticsContext: PropTypes.func,
};

const noop = () => [];

// eslint-disable-next-line @repo/internal/react/no-class-components
class AnalyticsListener extends Component<Props> {
  static contextTypes = ContextTypes;
  static childContextTypes = ContextTypes;

  contextValue: AnalyticsReactContextInterface;

  constructor(props: Props) {
    super(props);

    this.contextValue = {
      getAtlaskitAnalyticsContext: this.getAtlaskitAnalyticsContext,
      getAtlaskitAnalyticsEventHandlers: this.getAnalyticsEventHandlers,
    };
  }

  getChildContext = () => ({
    getAtlaskitAnalyticsEventHandlers: this.getAnalyticsEventHandlers,
    getAtlaskitAnalyticsContext: this.getAtlaskitAnalyticsContext,
  });

  getAnalyticsEventHandlers = () => {
    const { channel, onEvent } = this.props;
    const { getAtlaskitAnalyticsEventHandlers = noop } = this.context;

    const handler: UIAnalyticsEventHandler = (event, eventChannel) => {
      if (channel === '*' || channel === eventChannel) {
        onEvent(event, eventChannel);
      }
    };

    return [handler, ...getAtlaskitAnalyticsEventHandlers()];
  };

  getAtlaskitAnalyticsContext = () => {
    const { getAtlaskitAnalyticsContext = noop } = this.context;
    return getAtlaskitAnalyticsContext();
  };

  render() {
    const { children } = this.props;
    return (
      <AnalyticsReactContext.Provider value={this.contextValue}>
        {children}
      </AnalyticsReactContext.Provider>
    );
  }
}

export default AnalyticsListener;
