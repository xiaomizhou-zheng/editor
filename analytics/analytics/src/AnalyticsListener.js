/* eslint-disable react/sort-comp */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import matchEvent from './matchEvent';

/*
The Listener component is responsible for calling its `onEvent` handler when a
child component fires an analytics event, and passing the event up the hierarchy
*/
const ContextTypes = {
  onAnalyticsEvent: PropTypes.func,
};

if (process.env.NODE_ENV !== 'production' && !process.env.CI) {
  // eslint-disable-next-line no-console
  console.warn(
    'The @atlaskit/analytics package has been deprecated. Please use the @atlaskit/analytics-next package instead.',
  );
}

class AnalyticsListener extends Component {
  static defaultProps = {
    match: '*',
    matchPrivate: false,
  };

  static contextTypes = ContextTypes;

  static childContextTypes = ContextTypes;

  getChildContext() {
    return {
      onAnalyticsEvent: this.onAnalyticsEvent,
    };
  }

  onAnalyticsEvent = (name, data, isPrivate) => {
    // Call this component's onEvent method if it's a match
    // eslint-disable-next-line react/prop-types
    const { onEvent, match, matchPrivate } = this.props;
    if (
      matchPrivate === isPrivate &&
      matchEvent(match, name) &&
      typeof onEvent === 'function'
    ) {
      // send a clean data object so it can't be mutated between listeners
      const eventData = { ...data };
      onEvent(name, eventData);
    }

    // Pass the event up the hierarchy
    const { onAnalyticsEvent } = this.context;
    if (typeof onAnalyticsEvent === 'function') {
      onAnalyticsEvent(name, data, isPrivate);
    }
  };

  render() {
    const { children } = this.props; // eslint-disable-line react/prop-types
    return React.Children.only(children);
  }
}

export default AnalyticsListener;
