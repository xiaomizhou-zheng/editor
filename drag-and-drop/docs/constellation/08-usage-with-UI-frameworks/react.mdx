---
order: 1
title: Usage with UI Frameworks
description: How to use @atlaskit/drag-and-drop with different UI libraries and frameworks
---

### Attaching behaviour

We advise that you attach drag and drop behaviour to elements using [the `useEffect` hook](https://reactjs.org/docs/hooks-effect.html).

```tsx
// card.tsx
import {draggable, dropTargetForElements} from '@atlaskit/drag-and-drop/adapter/element';

export default function Card({ item }: { item: Item }) {
  const ref = useRef<HTMLDivElement | null>(null);
  const itemId = item.itemId;
  const [state, setState] = useState<DraggableState>('idle');

  useEffect(() => {
    const cleanup = combine(
      draggable({
        element: ref.current,
        getInitialData: () => ({ type: 'card', itemId: itemId }),
      }),
      dropTargetForElements({
        element: ref.current,
        canDrop: args => args.source.data.type === 'card',
      }),
    );
    return cleanup;
  }, [itemId]);

  return (
    <div ref={ref}>
      item id: {item.itemId}</span>
    </div>
  );
};
```

It is fine if your `draggable` or `dropTargetForElements` effect is cleaned up and re-created, even during a drag. For both of `draggable` and _drop targets_, the `element` is used as the _key_ for the entity. For more information, see [entity equality](TODO);

### Monitors

Using effects is also a nice way to manage the lifecyle of monitors:

```tsx
export function App() {

  useEffect(() => {
    // this is nice as the monitor will be cleaned up when <App /> is unmounted
    const cleanup = monitorForFiles({
      onDragStart: () => console.log('A file is dragging!');
    });

    return cleanup;
  }, []);

  return <Example>;
};
```

### Deferred loading

`react` lazy loading is based on components ([more details](https://reactjs.org/docs/code-splitting.html)).

You can use this component pattern with `@atlaskit/drag-and-drop`:

```tsx
import {Suspense, lazy} from 'react';

// importing our `card.tsx` file from above
const LazyCard = lazy(() => ('./card.tsx'));

function App() {
  return <Suspense fallback="loading..."><LazyCard></Suspense>
}
```

As `@atlaskit/drag-and-drop` is not tied to `react`, you can also attach drag and drop behaviour at some point after a component has already been rendered:

```tsx
function Card({ item }: { item: Item }) {
  const ref = useRef<HTMLDivElement | null>(null);
  const itemId = item.itemId;

  useEffect(() => {
    const controller = new AbortController();

    // Look! We are attaching behaviour after the component has rendered!
    // Note: Atlassian's, please use http://go/react-async rather than this promise based approach
    (async () => {
      const modules = await Promise.all([
        await import('@atlaskit/drag-and-drop/adapter/element'),
        await import('@atlaskit/drag-and-drop/util/combine'),
      ]);

      if (controller.signal.aborted) {
        return;
      }

      const [{ draggable, dropTargetForElements }, { combine }] = modules;

      if (!ref.current) {
        return;
      }

      const cleanup = combine(
        draggable({
          element: ref.current,
          getInitialData: () => ({ type: 'card', itemId: itemId }),
        }),
        dropTargetForElements({
          element: ref.current,
          canDrop: args => args.source.data.type === 'card',
          getData: () => ({ type: 'card', itemId: itemId }),
        }),
      );

      controller.signal.addEventListener('abort', cleanup, { once: true });
    })();

    return () => {
      controller.abort();
    };
  }, [itemId]);

  return (
    <div ref={ref}>
      item id: {item.itemId}</span>
    </div>
  );
};
```

For more details, see our [deferred loading guide](/components/drag-and-drop/deferred-loading).
