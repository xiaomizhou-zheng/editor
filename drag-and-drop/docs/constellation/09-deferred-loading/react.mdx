---
order: 1
title: Deferred loading
description: Improving performing by delaying the loading of drag and drop
---

You can use _dynamic imports_ to do things like deferring the import of a module until a `react` `useEffect`.

## Basic usage

```tsx
import React, {useEffect} from 'react';
import invariant from 'tiny-invariant';

function Card() {
  const ref = useRef<HTMLDivElement | null>(null);
  const [state, setState] = useState();

  useEffect(() => {
    const controller = new AbortController();

    (async () => {
      // load in all the modules that you need
      const modules = await Promise.all([
        import('@atlaskit/drag-and-drop/adapter/element'),
        import('@atlaskit/drag-and-drop/util/combine'),
      ]);
      if (controller.signal.aborted) {
        return;
      }
      invariant(el);

      const [{draggable, dropTargetForElements}, {combine}] = modules;

      const cleanup = combine(
        draggable({
        element: el,
        onDragStart: () => setState('dragging'),
        onDrop: () => setState('idle'),
      }),
      dropTargetForElements({
        element: el,
        onDrop: () => console.log('dropped on')
      }));

      controller.signal.addEventListener('abort', cleanup, { once: true });
    })();

    return () => {
      controller.abort();
    };
  }, []);

  return <div ref={ref}>Drag me<div>
});
```

## Maximise Deferred

Rather than importing all the modules you need directly, you can pull all your imports and logic out into a seperate file in order to defer more code.

```ts
// attach.js
// regular imports
import {
  draggable,
  dropTargetForElements,
} from '@atlaskit/drag-and-drop/adapter/element';
import { combine } from '@atlaskit/drag-and-drop/util/combine';
import invariant from 'tiny-invariant';

export function attach({ ref, setState }) {
  // all of this code can get deferred
  const el = ref.current;
  invariant(el);
  return combine(
    draggable({
      element: el,
      onDragStart: () => setState('dragging'),
      onDrop: () => setState('idle'),
    }),
    dropTargetForElements({
      element: el,
      onDrop: () => console.log('dropped on'),
    }),
  );
}
```

```tsx
import React, {useEffect} from 'react';

function Card() {
  const ref = useRef<HTMLDivElement | null>(null);
  const [state, setState] = useState();

  useEffect(() => {
    const controller = new AbortController();

    (async () => {
      // just loading in the single module that we need
      const { attach } = import('./attach');
      if (controller.signal.aborted) {
        return;
      }
      const cleanup = attach({setState, ref});

      controller.signal.addEventListener('abort', cleanup, { once: true });
    })();

    return () => {
      controller.abort();
    };
  }, []);

  return <div ref={ref}>Drag me<div>
});
```
