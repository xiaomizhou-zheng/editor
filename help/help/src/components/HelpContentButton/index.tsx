import React, { useCallback } from 'react';
import ShortcutIcon from '@atlaskit/icon/glyph/shortcut';
import Tooltip from '@atlaskit/tooltip';
import { NotificationIndicator } from '@atlaskit/notification-indicator';
import { NotificationLogProvider } from '@atlaskit/notification-log-client';
import {
  useAnalyticsEvents,
  UIAnalyticsEvent,
  AnalyticsContext,
} from '@atlaskit/analytics-next';

import {
  name as packageName,
  version as packageVersion,
} from '../../version.json';

import {
  HelpContentButtonContainer,
  HelpContentButtonIcon,
  HelpContentButtonText,
  HelpContentButtonExternalLinkIcon,
  HelpContentButtonExternalNotificationIcon,
} from './styled';

export type Props = {
  id?: string;
  href?: string;
  notificationMax?: number;
  notificationLogProvider?: Promise<NotificationLogProvider>;
  text: string;
  icon?: React.ReactChild;
  tooltipText?: string;
  onClick?: (
    id: string,
    analytics: UIAnalyticsEvent,
    event: React.MouseEvent<HTMLElement>,
  ) => void;
};

const analitycsContextData = {
  componentName: 'HelpContentButton',
  packageName,
  packageVersion,
};

const HelpContentButton = ({
  id = '',
  href,
  notificationMax = 3,
  notificationLogProvider,
  text,
  icon,
  onClick,
  tooltipText,
}: Props) => {
  const { createAnalyticsEvent } = useAnalyticsEvents();

  const handleOnClick = useCallback(
    (event: React.MouseEvent<HTMLElement>) => {
      if (onClick) {
        const analyticsEvent = createAnalyticsEvent({
          action: 'clicked',
        });

        onClick(id, analyticsEvent, event);
      }
    },
    [createAnalyticsEvent, id, onClick],
  );

  const buttonContent = (
    <>
      <HelpContentButtonIcon>{icon}</HelpContentButtonIcon>
      <HelpContentButtonText>
        {text}
        {notificationLogProvider !== null && (
          <HelpContentButtonExternalNotificationIcon>
            <NotificationIndicator
              notificationLogProvider={notificationLogProvider}
              max={notificationMax}
              appearance="primary"
            />
          </HelpContentButtonExternalNotificationIcon>
        )}
        {href != null && (
          <HelpContentButtonExternalLinkIcon data-testid="shortcutIcon">
            <ShortcutIcon size="small" label="" />
          </HelpContentButtonExternalLinkIcon>
        )}
      </HelpContentButtonText>
    </>
  );

  return (
    <AnalyticsContext data={analitycsContextData}>
      <HelpContentButtonContainer
        onClick={handleOnClick}
        href={href}
        id={id}
        target={href && '_blank'}
      >
        {tooltipText ? (
          <Tooltip content={tooltipText} position="left">
            {buttonContent}
          </Tooltip>
        ) : (
          <>{buttonContent}</>
        )}
      </HelpContentButtonContainer>
    </AnalyticsContext>
  );
};

export default HelpContentButton;
