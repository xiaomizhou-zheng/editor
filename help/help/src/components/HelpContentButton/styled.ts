/** @jsx jsx */

import styled from 'styled-components';
import * as colors from '@atlaskit/theme/colors';
import { gridSize } from '@atlaskit/theme/constants';
import { token } from '@atlaskit/tokens';

export const HelpContentButtonContainer = styled.a`
  display: block;
  cursor: pointer;
  width: calc(100% - ${gridSize() * 2}px);
  color: ${token('color.text.subtle', colors.N600)};
  padding: ${gridSize}px;
  border-radius: 3px;

  &:hover,
  &:focus,
  &:visited,
  &:active {
    text-decoration: none;
    outline: none;
    outline-offset: none;
    color: ${token('color.text.subtle', colors.N600)};
  }

  &:focus {
    box-shadow: ${token('color.border.focused', colors.B100)} 0 0 0 2px inset;
  }

  &:hover {
    background-color: ${token(
      'color.background.neutral.subtle.hovered',
      colors.N30,
    )};
  }

  &:active {
    background-color: ${token(
      'color.background.neutral.subtle.pressed',
      colors.B50,
    )};
  }
`;

export const HelpContentButtonIcon = styled.div`
  display: inline-block;
  vertical-align: middle;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  position: relative;

  & span {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
`;

export const HelpContentButtonText = styled.div`
  width: calc(100% - 20px);
  display: inline-block;
  vertical-align: middle;
  padding: 0 ${gridSize}px;
  box-sizing: border-box;
`;

export const HelpContentButtonExternalLinkIcon = styled.div`
  display: inline-block;
  vertical-align: middle;
  padding-left: ${gridSize() / 2}px;
`;

export const HelpContentButtonExternalNotificationIcon = styled.div`
  display: inline-block;
  vertical-align: middle;
  margin-top: -${gridSize() / 2}px;
  padding-left: ${gridSize() / 2}px;
`;
