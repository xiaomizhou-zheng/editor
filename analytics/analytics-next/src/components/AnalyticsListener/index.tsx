import LegacyAnalyticsListener from './LegacyAnalyticsListener';
import ModernAnalyticsListener from './ModernAnalyticsListener';
import { AnalyticsListenerFunction } from './types';

let ExportedAnalyticsListener: AnalyticsListenerFunction;
if (
  typeof process !== 'undefined' &&
  process !== null &&
  process.env?.['ANALYTICS_NEXT_MODERN_CONTEXT']
) {
  ExportedAnalyticsListener = ModernAnalyticsListener as any;
} else {
  ExportedAnalyticsListener = LegacyAnalyticsListener as any;
}

export default ExportedAnalyticsListener;
