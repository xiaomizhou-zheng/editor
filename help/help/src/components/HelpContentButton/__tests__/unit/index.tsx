import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import ShipIcon from '@atlaskit/icon/glyph/ship';
import { NotificationLogClient } from '@atlaskit/notification-log-client';
import AnalyticsListener from '@atlaskit/analytics-next/AnalyticsListener';

import HelpContentButton from '../../index';

const mockOnClick = jest.fn();
const analyticsSpy = jest.fn();
const buttonLabel = 'Test Content Button';
const notificationNumber = 5;

// Mockup notification Promise
class MockNotificationLogClient extends NotificationLogClient {
  notificationCounter: number;
  constructor(notificationCounter: number = notificationNumber) {
    super('', '');
    this.notificationCounter = notificationCounter;
  }

  public async countUnseenNotifications() {
    return Promise.resolve({ count: this.notificationCounter });
  }
}

describe('HelpContentButton', () => {
  // FIXME This snapshot covers a large amount of unrelated DOM content
  // and asserts on changes that are unrelated to this component
  //
  // Consider:
  // 1. Mocking upstream deps to narrow the snapshot
  // 2. What this snapshot is trying to assert on and narrow the test
  // 3. Delete the test
  it.skip('Should match snapshot', async () => {
    const notificationsClient = new MockNotificationLogClient();
    const notificationLogProvider = Promise.resolve(notificationsClient);
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          href="https://www.atlassian.com/"
          notificationMax={9}
          notificationLogProvider={notificationLogProvider}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { container } = render(component);

    await waitFor(() => notificationLogProvider);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should display the notification icon with the number returned by the notificationLogProvider', async () => {
    const notificationsClient = new MockNotificationLogClient(5);
    const notificationLogProvider = Promise.resolve(notificationsClient);
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          href="https://www.atlassian.com/"
          notificationMax={9}
          notificationLogProvider={notificationLogProvider}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { queryByText } = render(component);

    await waitFor(() => notificationLogProvider);

    const notificationCounterElm = queryByText(`${notificationNumber}`);
    expect(notificationCounterElm).not.toBeNull();
  });

  it(`Should display the notification icon with the number specified in the prop notificationMax
   followed by a "+" when the number returned by the notificationLogProvider if higher than the
   number specified in notificationMax`, async () => {
    const notificationsClient = new MockNotificationLogClient(20);
    const notificationLogProvider = Promise.resolve(notificationsClient);
    const notificationMax = 9;
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          href="https://www.atlassian.com/"
          notificationMax={notificationMax}
          notificationLogProvider={notificationLogProvider}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { queryByText } = render(component);

    await waitFor(() => notificationLogProvider);

    const notificationCounterElm = queryByText(`${notificationMax}+`);
    expect(notificationCounterElm).not.toBeNull();
  });

  it(`Should not display the notification icon when notificationLogProvider is not defined`, async () => {
    const notificationMax = 9;
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          href="https://www.atlassian.com/"
          notificationMax={notificationMax}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { queryByText } = render(component);

    const notificationCounterElm = queryByText(`${notificationNumber}`);
    expect(notificationCounterElm).not.toBeInTheDocument();
  });

  it(`Should not display the notification icon when notificationLogProvider is not defined`, async () => {
    const notificationMax = 9;
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          href="https://www.atlassian.com/"
          notificationMax={notificationMax}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { queryByText } = render(component);

    const notificationCounterElm = queryByText(`${notificationNumber}`);
    expect(notificationCounterElm).toBeNull();
  });

  it(`Should display the Shortcut icon if the "href" prop is defined`, async () => {
    const notificationMax = 9;
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          href="https://www.atlassian.com/"
          notificationMax={notificationMax}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { queryByTestId } = render(component);

    const shortcutIcon = queryByTestId('shortcutIcon');
    expect(shortcutIcon).not.toBeNull();
  });

  it(`Should not display the Shortcut icon if the "href" is not defined`, async () => {
    const notificationMax = 9;
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          notificationMax={notificationMax}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { queryByTestId } = render(component);

    const shortcutIcon = queryByTestId('shortcutIcon');
    expect(shortcutIcon).toBeNull();
  });

  it('Should call handleOnClick when the user click the button', () => {
    const notificationMax = 9;
    const component = (
      <AnalyticsListener channel="help" onEvent={analyticsSpy}>
        <HelpContentButton
          id="testHelpContentButton"
          notificationMax={notificationMax}
          text={buttonLabel}
          icon={<ShipIcon size="medium" label="" />}
          onClick={mockOnClick}
        />
      </AnalyticsListener>
    );
    const { queryByText } = render(component);

    const button = queryByText(buttonLabel);
    expect(button).not.toBeNull();

    if (button) {
      fireEvent.click(button);
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    }
  });
});
