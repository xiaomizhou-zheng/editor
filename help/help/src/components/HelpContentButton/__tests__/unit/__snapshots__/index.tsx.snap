// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpContentButton Should match snapshot 1`] = `
.emotion-0 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
  width: 24px;
  height: 24px;
}

.emotion-0 > svg {
  overflow: hidden;
  pointer-events: none;
  max-width: 100%;
  max-height: 100%;
  color: var(--icon-primary-color);
  fill: var(--icon-secondary-color);
  vertical-align: bottom;
}

.emotion-0 > svg stop {
  stop-color: currentColor;
}

@media screen and (forced-colors:active) {
  .emotion-0 > svg {
    -webkit-filter: grayscale(1);
    filter: grayscale(1);
    --icon-primary-color: CanvasText;
    --icon-secondary-color: Canvas;
  }
}

.emotion-0 > svg {
  width: 24px;
  height: 24px;
}

.emotion-3 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
  width: 16px;
  height: 16px;
}

.emotion-3 > svg {
  overflow: hidden;
  pointer-events: none;
  max-width: 100%;
  max-height: 100%;
  color: var(--icon-primary-color);
  fill: var(--icon-secondary-color);
  vertical-align: bottom;
}

.emotion-3 > svg stop {
  stop-color: currentColor;
}

@media screen and (forced-colors:active) {
  .emotion-3 > svg {
    -webkit-filter: grayscale(1);
    filter: grayscale(1);
    --icon-primary-color: CanvasText;
    --icon-secondary-color: Canvas;
  }
}

.emotion-3 > svg {
  width: 16px;
  height: 16px;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding-block: 2px;
  padding-inline: 6px;
  border-radius: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 12px;
  line-height: 12px;
  text-align: center;
}

<a
  class="sc-bdVaJa jbwMKo"
  href="https://www.atlassian.com/"
  id="testHelpContentButton"
  target="_blank"
>
  <div
    class="sc-bwzfXH WTOuK"
  >
    <span
      aria-hidden="true"
      class="emotion-0"
      role="presentation"
      style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
    >
      <svg
        height="24"
        role="presentation"
        viewBox="0 0 24 24"
        width="24"
      >
        <g
          fill="currentColor"
          fill-rule="evenodd"
        >
          <path
            d="M6 12h8v-2H6v2zM4 8.99C4 8.445 4.456 8 5.002 8h9.996C15.55 8 16 8.451 16 8.99V14H4V8.99z"
            fill-rule="nonzero"
          />
          <path
            d="M6 7.005C6 5.898 6.898 5 7.998 5h2.004C11.106 5 12 5.894 12 7.005V10H6V7.005zm4 0V7H7.999c.005 0 .002.003.002.005V8h2v-.995z"
          />
          <path
            d="M4.5 17h13.994l1.002-3H4.14l.36 3zm-2.495-4.012A.862.862 0 012.883 12h18.393c.55 0 .857.417.681.944l-1.707 5.112c-.174.521-.758.944-1.315.944H3.725a1.149 1.149 0 01-1.118-.988l-.602-5.024z"
            fill-rule="nonzero"
          />
        </g>
      </svg>
    </span>
  </div>
  <div
    class="sc-htpNat iUSDkM"
  >
    Test Content Button
    <div
      class="sc-ifAKCX dTKZCs"
    >
      <div
        data-test-selector="NotificationIndicator"
      >
        <span
          class="emotion-2"
        >
          <span
            class="emotion-1"
          >
            5
          </span>
        </span>
      </div>
    </div>
    <div
      class="sc-bxivhb crZlKE"
      data-testid="shortcutIcon"
    >
      <span
        aria-hidden="true"
        class="emotion-3"
        role="presentation"
        style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
      >
        <svg
          height="24"
          role="presentation"
          viewBox="0 0 24 24"
          width="24"
        >
          <g
            fill="currentColor"
          >
            <path
              d="M19.005 19c-.003 0-.005.002-.005.002l.005-.002zM5 19.006c0-.004-.002-.006-.005-.006H5v.006zM5 4.994V5v-.006zM19 19v-6h2v6.002A1.996 1.996 0 0119.005 21H4.995A1.996 1.996 0 013 19.006V4.994C3 3.893 3.896 3 4.997 3H11v2H5v14h14zM5 4.994V5v-.006zm0 14.012c0-.004-.002-.006-.005-.006H5v.006zM11 5H5v14h14v-6h2v6.002A1.996 1.996 0 0119.005 21H4.995A1.996 1.996 0 013 19.006V4.994C3 3.893 3.896 3 4.997 3H11v2zm8 0v3a1 1 0 002 0V4a1 1 0 00-1-1h-4a1 1 0 000 2h3z"
            />
            <path
              d="M12.707 12.707l8-8a1 1 0 10-1.414-1.414l-8 8a1 1 0 001.414 1.414z"
            />
          </g>
        </svg>
      </span>
    </div>
  </div>
</a>
`;
